### ISSUE TITLE
Send continuous output on axis end?

### ISSUE THREAD

**Issue by Scoobydoomed** [WhiteMagic/JoystickGremlin#80]
Hi,
Is it possible to  have my pedal send continuous output when fully deflected? 
The problem I am trying to solve is that the pedal only sends output when a change is made to the physical device, so for instance if my ship in star citizen spawns while i have my foot on the pedal it will not move until i let go of the pedal and push it again. I imagine this could be a limitation of the device but i was thinking if it would maybe possible to do this somehow using macro? 
i tried a few things (sending slider output or keyboard output, condition device axis between 0.99 and 1.00, playing with toggle and hold etc) but nothing seems to work. 

I am using vkb pedals if that makes any difference. 

TIA

**Comment by Scoobydoomed** [comment]
managed to get half way.

i can get a button to spam once i am at max deflection but i cant get it to stop when leaving the activation zone. 
![1](https://user-images.githubusercontent.com/35772381/35355550-4e249002-0156-11e8-85a9-49c153049bd0.JPG)

![2](https://user-images.githubusercontent.com/35772381/35355556-50d77d14-0156-11e8-95f7-ade98d931a46.JPG)

![3](https://user-images.githubusercontent.com/35772381/35355559-5432f1b4-0156-11e8-9dc6-c9db434719f6.JPG)

**Comment by WhiteMagic** [comment]
The pedal not sending max deflection value once it hit it is by design, as joystick like devices send information only when a change occurs which in your case doesn't happen at max deflection.

I think the solution that has the best chance of doing what you want would be to have a virtual button setup for 0.99 to 1.0 or 1.0 to 1.0 (not sure what Gremlin will make of the last range tbh) and then have a macro that sends max deflection values to the vJoy device you want to use for that pedal axis. The macro might need to send two slightly different values for SC to recognize the change, say 0.999 and 1.0. That macro should be set to repeat on "hold", so for as long as you "hold the virtual button down" which corresponds to you depressing the pedal fully.

The reason your macro wouldn't have done much is that it would repeatedly send a "key down" event for the T key but this would not actually be seen as sending another T key stroke. Repeated keys, i.e. the spamming of TTTTTTTTTTTTTTTTT is something the operating system produces when it sees that a key is being held down for a certain duration and then interprets this as the user wanting to send the same input at a fixed interval. I think games typically don't care for this and simply track the state of a key, i.e. whether or not it's down or up.

The reason the macro isn't behaving the way you intend is a bit of a mix. The "toggle" repeat mode will start executing a macro when the corresponding "button" is pressed for the first time and will stop executing the macro when the "button" is pressed the second time. The issue with that is that stopping your "Press T" macro won't cause it to send a "Release T" event. I'm not sure if the macro you have configured for the "release virtual button" case ever would get run with a toggle repeat macro, that's something I have to look at. In general toggle repeat macros and conditions have the oddest interactions.

I'll have a look at the macro thing later today but I think the solution I outlined above with the macro which sends two vJoy axis settings is probably the cleanest solution. I've in the past used similar tricks in SC to get rid of the max throttle thing when you come out of quantum travel.

**Comment by Scoobydoomed** [comment]
Thanks for the replay WhiteMagic,

I tried as you suggested using the "hold" option but my results are the same. The button starts spamming when i press my pedal but it will not stop spamming even after leaving the virtual button zone. In fact the button keeps spamming even after stopping and starting the profile again in JG and only stops while the profile is not active, or i restart JG.  

here are screens of my last attempt:
![1](https://user-images.githubusercontent.com/35772381/35409099-ec82fc66-0219-11e8-97ec-6086ecf627b2.JPG)
![2](https://user-images.githubusercontent.com/35772381/35409110-f0984612-0219-11e8-8a3c-79d63fa07f6e.JPG)
![3](https://user-images.githubusercontent.com/35772381/35409115-f2eb2d1c-0219-11e8-9380-1499f1ee8a08.JPG)

**Comment by WhiteMagic** [comment]
I just played around with what should work, essentially hold macro that send vJoy axis inputs when the physical axis is at maximum deflection. This works, however, the macro will never stop which is due to bug(s) in Gremlin with regards to macros triggered by virtual axis buttons.

The good news is that I've fixed those a while ago in the development version, the bad news is that it probably will take another week or two before I'm done with the next version.

In any case the below setup that should work, with the exception of the macro never terminating with R9.2. That macro would be placed on the physical axis while the vJoy outputs are the vJoy axis you want to control.
![cont_example](https://user-images.githubusercontent.com/5612/35464073-2d4b2226-0348-11e8-88e2-001c0ae29741.png)
